nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/PeerDescriptionModifier.java

```diff
@@ -0,0 +1,165 @@
+        private SiteToSiteTransportProtocol protocol;
+        private PreparedQuery predicate;
+        private PreparedQuery hostname;
+        private PreparedQuery port;
+        private PreparedQuery secure;
+
+        private Route validate() {
+            if (hostname == null) {
+                throw new IllegalArgumentException(
+                        format("Found an invalid Site-to-Site route definition [%s] 'hostname' is not specified.", name));
+            }
+            if (port == null) {
+                throw new IllegalArgumentException(
+                        format("Found an invalid Site-to-Site route definition [%s] 'port' is not specified.", name));
+            }
+            return this;
+        }
+
+        private PeerDescription getTarget(final Map<String, String> variables) {
+            final String targetHostName = hostname.evaluateExpressions(variables, null);
+            if (isBlank(targetHostName)) {
+                throw new IllegalStateException("Target hostname was not resolved for the route definition " + name);
+            }
+
+            final String targetPortStr = port.evaluateExpressions(variables, null);
+            if (isBlank(targetPortStr)) {
+                throw new IllegalStateException("Target port was not resolved for the route definition " + name);
+            }
+
+            final String targetIsSecure = secure == null ? null : secure.evaluateExpressions(variables, null);
+            return new PeerDescription(targetHostName, Integer.valueOf(targetPortStr), Boolean.valueOf(targetIsSecure));
+        }
+    }
+
+    private Map<SiteToSiteTransportProtocol, List<Route>> routes;
+
+
+    private static final String PROPERTY_PREFIX = "nifi.remote.route.";
+
+    public PeerDescriptionModifier(final NiFiProperties properties) {
+        final Map<String, List<String>> routeDefinitions = properties.getPropertyKeys().stream()
+                .filter(propertyKey -> propertyKey.startsWith(PROPERTY_PREFIX))
+                .collect(Collectors.groupingBy(propertyKey -> propertyKey.substring(PROPERTY_PREFIX.length(), propertyKey.lastIndexOf('.'))));
+
+        routes = routeDefinitions.entrySet().stream().map(routeDefinition -> {
+            final Route route = new Route();
+            // E.g. raw.example1, http.example2
+            final String[] protocolAndRoutingName = routeDefinition.getKey().split("\\.");
+            route.protocol = SiteToSiteTransportProtocol.valueOf(protocolAndRoutingName[0].toUpperCase());
+            route.name = protocolAndRoutingName[1];
+            routeDefinition.getValue().forEach(propertyKey -> {
+                final String routingConfigName = propertyKey.substring(propertyKey.lastIndexOf('.') + 1);
+                final String routingConfigValue = properties.getProperty(propertyKey);
+                switch (routingConfigName) {
+                    case "when":
+                        route.predicate = Query.prepare(routingConfigValue);
```

Comment:
I also could not make `prepare` throw but the JavaDocs state that it can. Could we add a try/catch around this switch to gracefully handle if it does throw?